sasl_bind.c

```diff
@@ -0,0 +1,21 @@
+#include <ldap.h>
+#include "sasl_bind.h"
+
+static int
+sasl_trivial_interact (LDAP * ld, unsigned flags, void *defaults,
+                       void *sasl_interact)
+{
+  (void) ld;
+  (void) flags;
+  (void) defaults;
+  (void) sasl_interact;
+  return LDAP_SUCCESS;
+}
+
+int
+trivial_external_sasl_bind (LDAP * ld)
+{
+  return ldap_sasl_interactive_bind_s (ld, NULL, "EXTERNAL", NULL, NULL,
```

Comment:
I don't remember why I used `ldap_sasl_interactive_bind_s()` instead of `ldap_sasl_bind_s()`; I probably failed to achieve anything with the latter. [Others](https://adam.younglogic.com/2012/02/exteranl-sasl/) also struggled with this and used a different interaction callback, which actually does something. I wonder what and why. It would be good to find some documentation on the matter.